time/CalendarWidget.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  CalendarWidget.swift
+//  seximal
+//
+//  Created by Henrik Storch on 24.06.21.
+//
+
+import SwiftUI
+import Intents
+import WidgetKit
+
+
+struct CalendarWidget: Widget {
+    let kind = "cal"
+    
+    var body: some WidgetConfiguration {
+        IntentConfiguration(kind: kind, intent: CalendarIntent.self, provider: CalendarProvider()) { entry in
+            CalendarEntryView(entry: entry)
+        }
+        .configurationDisplayName("Calendar Widget")
+        .description("A widget showing the current month and date in a seximal calendar.")
+    }
+}
+
+struct CalendarEntry: TimelineEntry {
+    let date: Date
+    let configuration: CalendarIntent
+}
+
+struct CalendarProvider: IntentTimelineProvider {
+    
+    func placeholder(in context: Context) -> CalendarEntry {
+        CalendarEntry(date: Date(), configuration: CalendarIntent())
+    }
+    
+    func getSnapshot(for configuration: CalendarIntent, in context: Context, completion: @escaping (CalendarEntry) -> Void) {
+        let entry = CalendarEntry(date: Date(), configuration: configuration)
+        completion(entry)
+    }
+    
+    func getTimeline(for configuration: CalendarIntent, in context: Context, completion: @escaping (Timeline<CalendarEntry>) -> Void) {
+        let midnight = Calendar.utc.startOfDay(for: Date())
+        let nextMidnight = Calendar.utc.date(byAdding: .day, value: 1, to: midnight)!
+        let next4Days = (1...3).map({ Calendar.utc.date(byAdding: .day, value: $0, to: midnight)! })
+        let entries = next4Days.map( { CalendarEntry(date: $0, configuration: configuration) })
+        let timeline = Timeline(entries: entries, policy: .after(nextMidnight))
```

Comment:
`policy` can probably be extended a bit, since it won't be different compared to e.g. the Weather.app